src/Orleans.Runtime/Transactions/TransactionAgent.cs

```diff
@@ -1,13 +1,99 @@
 ﻿using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Orleans.Runtime;
 using Orleans.Concurrency;
+using Orleans.Runtime.Configuration;
+using DateTime = System.DateTime;
 
 namespace Orleans.Transactions
 {
+    internal class TransactionAgentMetrics
+    {
+        //TPS in current monitor window
+        private const string BatchStartTransanctionsDeltaTPS = "TransactionAgent.BatchStartTransactions.Delta.TPS";
+        //avg latency in current monitor window
+        private const string AvgBatchStartTransactionsDeltaLatency = "TransactionAgent.BatchStartTransactions.Delta.AvgLatency";
+        private const string BatchCommitTransactionsDeltaTPS = "TransactionAgent.BatchCommitTransactions.Delta.TPS";
+        private const string AvgBatchCommitTransactionsDeltaLatency = "TransactionAgent.BatchCommitTransactions.Delta.AvgLatency";
+
+        internal int BatchStartTransactionsRequestCounter { get; set; }
+
+        internal int BatchCommitTransactionsRequestsCounter { get; set; }
+        private TimeSpan batchStartTransactionsRequestLatencyCounter = TimeSpan.Zero;
+        internal TimeSpan BatchStartTransactionsRequestLatencyCounter
+        {
+            get { return batchStartTransactionsRequestLatencyCounter; }
+            set
+            {
+                batchStartTransactionsRequestLatencyCounter = value;
+                this.periodicMonitor.TryAction(DateTime.UtcNow);
```

Comment:
May want to expose some sort of periotic trigger in the public surface and have it called in the TransactionAgents  ProcessRequests loop, rather than in property setters, which is definitely an unexpected behavior.

Reply:
I wanted to encapsulate metric related logic in this Metric class. So in that sense, metric reporting logic should be embedded in the class itself, without TransactionAgent/its monitoring target to worry about it. 
And setters are called during ProcessRequests loop, so functionality wise it is the same. 
But it is a bit weird to do it in setters, I will update this